### PR TITLE
Valgrind told me to do it

### PR DESCRIPTION
**Pull request by sethalves**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 43c3c621a8e9db8359a4a3c522de5275b53f3243 available at: https://gist.github.com/eed2998344fb8c768d47<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 26230cf493f35cfe53b39942eee8f23b0a17f003 available at: https://gist.github.com/eed2998344fb8c768d47<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4bf86bb824a800ad31f9be54b3a8b3322b1b535a available at: https://gist.github.com/eed2998344fb8c768d47<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7137b4137554605cd27d26faca9e6da8ba4f210b available at: https://gist.github.com/eed2998344fb8c768d47<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4527.dmg

**Comment by ZappoMan** [comment]
Can we get @jherico to review at least the line related to his code in this PR.

File: assignment-client/src/avatars/AvatarMixer.cpp [hunk 1]
```diff
@@ -169,7 +170,7 @@ void AvatarMixer::broadcastAvatarData() {
                     return true;
                 },
                 [&](const SharedNodePointer& otherNode) {
-                    AvatarMixerClientData* otherNodeData = otherNodeData = reinterpret_cast<AvatarMixerClientData*>(otherNode->getLinkedData());
+                    AvatarMixerClientData* otherNodeData = reinterpret_cast<AvatarMixerClientData*>(otherNode->getLinkedData());
```
Comment (ZappoMan) [inline assignment-client/src/avatars/AvatarMixer.cpp]: @jherico - can you look at this line please...

Comment (jherico) [inline assignment-client/src/avatars/AvatarMixer.cpp]: Yes, this was just a typo.  the change is fine.  Just to reassure, I would never include something like that double assignment without at least providing a comment as to why.

File: libraries/audio/src/AudioBuffer.h [hunk 2]
```diff
@@ -88,9 +88,9 @@ template< typename T >
 void AudioFrameBuffer< T >::deallocateFrames() {
     if (_frameBuffer) {
         for (uint32_t i = 0; i < _channelCountMax; ++i) {
-            delete _frameBuffer[i];
+            delete[] _frameBuffer[i];
```
Comment (jherico) [inline libraries/audio/src/AudioBuffer.h]: why aren't we just using a vector so that we don't have to do manual memory management?

**Pull request merged**